src/main/kotlin/com/dh/ondot/notification/domain/service/DeviceTokenService.kt

```
@@ -0,0 +1,57 @@
+    private val deviceTokenRepository: DeviceTokenRepository,
+) {
+
+    private val log = LoggerFactory.getLogger(javaClass)
+
+    @Transactional
+    fun registerOrUpdate(memberId: Long, fcmToken: String, deviceType: String) {
+        val existing = deviceTokenRepository.findByFcmToken(fcmToken)
+        if (existing == null) {
+            try {
+                deviceTokenRepository.save(DeviceToken.create(memberId, fcmToken, deviceType))
+            } catch (e: DataIntegrityViolationException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)